Documentation for the GitOps 1.11.0 release notes. (#68509)
openshift/openshift-docs: modules/gitops-release-notes-1-11-0.adoc

// Module included in the following assembly:
//
// * release_notes/gitops-release-notes.adoc

:_content-type: REFERENCE
[id="gitops-release-notes-1-11-0_{context}"]
= Release notes for {gitops-title} 1.11.0

{gitops-title} 1.11.0 is now available on {OCP} 4.12, 4.13, and 4.14.

[id="errata-updates-1-11-0_{context}"]
== Errata updates

[id="gitops-1-11-0-security-update-advisory_{context}"]
=== RHSA-2023:NNNN - {gitops-title} 1.11.0 security update advisory

Issued: 2023-12-07

The list of security fixes that are included in this release is documented in the following advisory:

* link:https://access.redhat.com/errata/RHSA-2023:NNNN[RHSA-2023:NNNN]

If you have installed the {gitops-title} Operator in the default namespace, to view the container images in this release, run the following command:

[source,terminal]
----
$ oc describe deployment gitops-operator-controller-manager -n openshift-gitops-operator

> ----

Reviewer: @masauso-lungu Do we have any security advidory this release? If not we can remove this section.

Please clarify with the SMEs. Thank you!

Reviewer 2: It's just a regular enhancement advisory this time!

Author: The available errata is an enhancemnt and i was due to update it. @reginapizza, do we need this RHEA published as we normally do with RHSA?, thank you

Reviewer 2: I'm not sure of the question @masauso-lungu... would this not be a question for @Srivaralakshmi ?

Reviewer: @masauso-lungu AFAIU, we only have a regular enhancement advisory this time! If there is no security advisory, then please drop this section. I hope this answer clarifies the confusion.

Thanks, @reginapizza for the response. This is helpful.

Author: Thank you @Srivaralakshmi . I will drop it.

[id="fixed-issues-1-11-0_{context}"]
== Fixed issues

> * Before this update, the Argo CD Notifications Controller did not support custom certificates added to `argocd-tls-certs-cm`. As a result, notification services with custom certificates did not receive notifications due to the "x509: certificate signed by unknown authority" error. This update fixes the issue by correctly initializing the cert resolver function in the Argo CD Notifications Controller to load all certificates stored in `argocd-tls-certs-cm`. Now, notification services with custom certificates can successfully receive notifications. link:https://issues.redhat.com/browse/GITOPS-2809[GITOPS-2809]

Reviewer: Suggestion: Rewrite for clarity. What is `argocd-tls-certs-cm`? A file or config map? Please mention that wherever it comes in content.

Reviewer 3: It's a configmap.

Author: ACK.

Resolved.

> * Before this update, users would face `PrometheusOperatorRejectedResources` alerts when the {gitops-title} Operator was not installed in the `openshift-gitops-operator` namespace. The problem affected users who upgraded from earlier versions of the {gitops-title} Operator to v1.10. This update fixes the issue by updating the operator's metrics service `serverName` to reflect the correct installation namespace. Now, users who upgrade or install the {gitops-title} Operator in namespaces other than `openshift-gitops-operator` should not see these alerts. link:https://issues.redhat.com/browse/GITOPS-3424[GITOPS-3424] 

Reviewer: ```suggestion
* Before this update, users would face `PrometheusOperatorRejectedResources` alerts when the {gitops-title} Operator was not installed in the `openshift-gitops-operator` namespace. The problem affected users who upgraded from earlier versions of the {gitops-title} Operator to v1.10. This update fixes the issue by updating the Operator's `serverName` metrics service to reflect the correct installation namespace. Now, users who upgrade or install the {gitops-title} Operator in namespaces other than `openshift-gitops-operator` should not see these alerts. link:https://issues.redhat.com/browse/GITOPS-3424[GITOPS-3424] 
```

Author: ACK.

Addressed.

// Module included in the following assembly:
//
// * release_notes/gitops-release-notes.adoc

> :_content-type: REFERENCE

Reviewer 4: See OpenShift documentation guidelines for [module file metadata](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#module-file-metadata)
```suggestion
:_mod-docs-content-type: REFERENCE
```

Author: ACK.

Addressed

[id="gitops-release-notes-1-11-0_{context}"]
= Release notes for {gitops-title} 1.11.0

{gitops-title} 1.11.0 is now available on {OCP} 4.12, 4.13, and 4.14.

[id="new-features-1-11-0_{context}"]
== New features

The current release adds the following improvement:

> * With this update, you can selectively disable the `redis` and `application-controller` components for an Argo CD instance in a specified namespace. These components are enabled by default.

Reviewer 5: This s not required as it was working as expected with openshift-gitops instance

> * With this update, you can selectively disable the `redis` and `application-controller` components for an Argo CD instance in a specified namespace. These components are enabled by default.

Author: Great. Thank you @varshab1210

> To disable a component, set the `enabled` flag to `false` in the `spec.<component>.enabled` field of the Argo CD Custom Resource (CR). link:https://issues.redhat.com/browse/GITOPS-3723[GITOPS-3723]

Reviewer 5: ```suggestion
To disable a component, set the `enabled` flag to `false` in the `.spec.<component>.enabled` field of the Argo CD Custom Resource (CR). link:https://issues.redhat.com/browse/GITOPS-3723[GITOPS-3723]
```

Author: ACK.
Addressed

+

> .Example: Disabling the `redis` and `application-controller` components

Reviewer 6: ```suggestion
For example:
```

Author: ACK.

Addressed

> [source, yaml]

Reviewer 4: Per OpenShift documentation [guidelines](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#code-blocks-command-syntax-and-example-output:~:text=do%20not%20insert%20a%20space):
> do not insert a space in between the [source] tag and the metadata.

```suggestion
[source,yaml]
```

Author: ACK.
Addressed

> ----

Reviewer 6: I suggest to consider adding initial fields to this YAML. For example, some more context like: `apiVersion: argoproj.io/v1alpha1
kind: ArgoCD
metadata:
  name: example-argocd`

Author: ACK.

Addressed.

spec:
  controller:
    enabled: false
  redis:
    enabled: false
----
+
[NOTE]
====

> This feature is currently limited to the `redis` and `application-controller` components. Support for other components is expected to be included in the forthcoming {gitops-title} v1.11.1 release.

Reviewer 4: See Red Hat supplementary style guide for [Future releases or plans](https://redhat-documentation.github.io/supplementary-style-guide/#statements-about-the-future)

> Do not promise that a feature or a fix for a known issue will be included in an upcoming release or according to a specific timeline.

```suggestion
This feature is currently limited to the `redis` and `application-controller` components. It is expected that support for other components will be included in a future {gitops-title} release.
```

> This feature is currently limited to the `redis` and `application-controller` components. Support for other components is expected to be included in the forthcoming {gitops-title} v1.11.1 release.

Author: ACK.
Addressed

====

[id="fixed-issues-1-11-0_{context}"]
== Fixed issues

The following issues have been resolved in the current release:

> * Before this update, the Argo CD Notifications Controller did not support custom certificates added to `argocd-tls-certs-cm` config map. As a result, notification services with custom certificates did not receive notifications due to the "x509: certificate signed by unknown authority" error. This update fixes the issue by correctly initializing the cert resolver function in the Argo CD Notifications Controller to load all certificates stored in `argocd-tls-certs-cm` config map. Now, notification services with custom certificates can successfully receive notifications. link:https://issues.redhat.com/browse/GITOPS-2809[GITOPS-2809]

Reviewer 6: ```suggestion
* Before this update, the Argo CD Notifications Controller did not support custom certificates added to the `argocd-tls-certs-cm` config map. As a result, notification services with custom certificates did not receive notifications due to the "x509: certificate signed by unknown authority" error. This update fixes the issue by correctly initializing the cert resolver function in the Argo CD Notifications Controller to load all certificates stored in the `argocd-tls-certs-cm` config map. Now, notification services with custom certificates can successfully receive notifications. link:https://issues.redhat.com/browse/GITOPS-2809[GITOPS-2809]
```

Author: ACK.
Addressed.

> * Before this update, the Argo CD Notifications Controller did not support custom certificates added to `argocd-tls-certs-cm` config map. As a result, notification services with custom certificates did not receive notifications due to the "x509: certificate signed by unknown authority" error. This update fixes the issue by correctly initializing the cert resolver function in the Argo CD Notifications Controller to load all certificates stored in `argocd-tls-certs-cm` config map. Now, notification services with custom certificates can successfully receive notifications. link:https://issues.redhat.com/browse/GITOPS-2809[GITOPS-2809]

Reviewer 4: ```suggestion
* Before this update, the Argo CD Notifications Controller did not support custom certificates added to `argocd-tls-certs-cm` config map. As a result, notification services with custom certificates did not receive notifications due to the `x509: certificate signed by unknown authority` error message. This update fixes the issue by correctly initializing the cert resolver function in the Argo CD Notifications Controller to load all certificates stored in `argocd-tls-certs-cm` config map. Now, notification services with custom certificates can successfully receive notifications. link:https://issues.redhat.com/browse/GITOPS-2809[GITOPS-2809]
```

> * Before this update, the Argo CD Notifications Controller did not support custom certificates added to `argocd-tls-certs-cm` config map. As a result, notification services with custom certificates did not receive notifications due to the "x509: certificate signed by unknown authority" error. This update fixes the issue by correctly initializing the cert resolver function in the Argo CD Notifications Controller to load all certificates stored in `argocd-tls-certs-cm` config map. Now, notification services with custom certificates can successfully receive notifications. link:https://issues.redhat.com/browse/GITOPS-2809[GITOPS-2809]

Author: ACK.
Addressed

> * Before this update, users would face `PrometheusOperatorRejectedResources` alerts when the {gitops-title} Operator was not installed in the `openshift-gitops-operator` namespace. The problem affected users who upgraded from earlier versions of the {gitops-title} Operator to v1.10. This update fixes the issue by updating the Operator's `serverName` metrics service to reflect the correct installation namespace. Now, users who upgrade or install the {gitops-title} Operator in namespaces other than `openshift-gitops-operator` should not see these alerts. link:https://issues.redhat.com/browse/GITOPS-3424[GITOPS-3424]

Reviewer 6: ```suggestion
* Before this update, users would face `PrometheusOperatorRejectedResources` alerts when the {gitops-title} Operator was not installed in the `openshift-gitops-operator` namespace. The problem affected users who upgraded from earlier versions of the {gitops-title} Operator to v1.10. This update fixes the issue by updating the Operator's `serverName` metrics service to reflect the correct installation namespace. Now, users who upgrade or install the {gitops-title} Operator in namespaces other than `openshift-gitops-operator` cannot see these alerts. link:https://issues.redhat.com/browse/GITOPS-3424[GITOPS-3424]
```

Reviewer 6: Can we use "cannot" instead of "should not"? I think "cannot" provides clarity

Author: IMO I feel "should not" sounds more natural and is more suitable for the described situation.

> 

Author: ACK.
addressed
